core/templates/fixed_vector.h

```diff
@@ -0,0 +1,153 @@
+/**************************************************************************/
+/*  fixed_vector.h                                                        */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+// A high speed Vector of fixed capacity.
+// Especially useful if you need to create an array on the stack, to
+// prevent dynamic allocations (especially in bottleneck code).
+
+// Choose CAPACITY such that it is enough for all elements that could be added through all branches.
+
+// Set force_trivial to true to gain a slight performance improvement on resize calls
+//  if the default value of the type can be ignored.
```

Comment:
Rework the comment above the class so it displays in full in VS Code when hovered:

```suggestion
/**
 * A high-speed Vector of fixed capacity.
 *
 * Especially useful if you need to create an array on the stack,
 * as it prevents dynamic allocations (especially in bottleneck code).
 *
 * Choose `CAPACITY` such that it is enough for all elements that could be added through all branches.
 * Set `force_trivial` to `true` to gain a slight performance improvement on resize calls
 * if the default value of the type can be ignored.
 */
```

![image](https://github.com/user-attachments/assets/d22b00c8-ced8-4ba9-a325-b026da2eca40)

Reply:
Good idea! We don't seem to do this often enough in the codebase as a whole :)